flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/source/TimestampsAndWatermarks.java

```diff
@@ -46,11 +45,14 @@ public interface TimestampsAndWatermarks<T> {
     /** Lets the owner/creator of the output know about latest emitted watermark. */
     @Internal
     interface WatermarkUpdateListener {
+
+        /** It should be called once the idle is changed. */
+        void updateIdle(boolean isIdle);
```

Comment:
I wonder if we need this API. Can't we tell by #updateCurrentEffectiveWatermark() already if the source is idle, e.g. checking if it is a MAX_WATERMARK?

Reply:
It seems by convention in Flink to mark idle separately rather than to update the aggregated watermark with max, so I think it makes sense to add this method.

Reply:
Actually, I have the same question during developing.

I add the method based on 2 reasons:

- The idle is related to watermark, so it's reasonable to add it.
- We can check if it's idle based on `MAX_WATERMARK`, however it's too complex. 
    - We send `MAX_WATERMARK` as the idle at the caller side
    - And check `MAX_WATERMARK` as the idle at the callee side

So I think adding the method is more clear.

Reply:
Sounds good to me, looks like the same reasoning previous implementers took